Content.Client/SurveillanceCamera/SurveillanceCameraNavMapControl.cs

```diff
@@ -0,0 +1,104 @@
+
+    private static readonly Color CameraActiveColor = Color.FromHex("#FF00FF");
+    private static readonly Color CameraInactiveColor = Color.FromHex("#a09f9fff");
+    private static readonly Color CameraSelectedColor = Color.FromHex("#fbff19ff");
+    private static readonly Color CameraInvalidColor = Color.FromHex("#fa1f1fff");
+
+    private readonly Texture _cameraTexture;
+    private string _activeCameraAddress = string.Empty;
+    private HashSet<string> _availableSubnets = new();
+    private (Dictionary<NetEntity, CameraMarker> Cameras, string ActiveAddress, HashSet<string> AvailableSubnets) _lastState;
+
+    public bool EnableCameraSelection { get; set; }
+
+    public event Action<NetEntity>? CameraSelected;
+
+
+    public SurveillanceCameraNavMapControl()
+    {
+        _cameraTexture = _resourceCache.GetTexture("/Textures/Interface/NavMap/beveled_triangle.png");
+
+        TrackedEntitySelectedAction += OnTrackedEntitySelected;
+    }
+
+    public void SetActiveCameraAddress(string address)
+    {
+        if (_activeCameraAddress == address)
+            return;
+
+        _activeCameraAddress = address;
+        ForceNavMapUpdate();
+    }
+
+    public void SetAvailableSubnets(HashSet<string> subnets)
+    {
+        if (_availableSubnets.SetEquals(subnets))
+            return;
+
+        _availableSubnets = subnets;
+        ForceNavMapUpdate();
+    }
+
+    protected override void UpdateNavMap()
+    {
+        base.UpdateNavMap();
+
+        if (MapUid is null || !_entityManager.TryGetComponent<SurveillanceCameraMapComponent>(MapUid, out var mapComp))
+            return;
+
+        var currentState = (mapComp.Cameras, _activeCameraAddress, _availableSubnets);
+        if (_lastState.Equals(currentState))
+            return;
+
+        _lastState = currentState;
+        UpdateCameraMarkers(mapComp);
+    }
+
+    private void UpdateCameraMarkers(SurveillanceCameraMapComponent mapComp)
+    {
+        TrackedEntities.Clear();
+
+        foreach (var (netEntity, marker) in mapComp.Cameras)
+        {
+            if (!_availableSubnets.Contains(marker.Subnet))
+                continue;
+
+            var coords = new EntityCoordinates(MapUid!.Value, marker.Position);
+
+            Color color;
+            if (string.IsNullOrEmpty(marker.Address))
+                color = CameraInvalidColor;
+            else if (marker.Address == _activeCameraAddress)
+                color = CameraSelectedColor;
+            else
+                color = marker.Active ? CameraActiveColor : CameraInactiveColor;
+
+            TrackedEntities[netEntity] = new NavMapBlip(
+                coords,
+                _cameraTexture,
+                color,
+                false,
+                EnableCameraSelection
+            );
+        }
+    }
+
+    private void OnTrackedEntitySelected(NetEntity? netEntity)
+    {
+        if (netEntity.HasValue && CameraSelected is not null)
+            CameraSelected(netEntity.Value);
```

Comment:
Can just do `CameraSelected?.Invoke(blah)` and at that point I'd just inline this method as well.